Automattic/studio
# Add sync pull remote site

- Related to https://github.com/Automattic/dotcom-forge/issues/9355

## Proposed Changes

- Add pull functionality to import a remote site

## Testing Instructions



- Run `STUDIO_SITE_SYNC=true npm start`
- Click on the Sync tab
- Connect a site
- Click on pull
- Wait until the process finishes
- Observe your site was imported correctly
- Repeat the process and change tabs and sites, observe the process continues correctly
- To reproduce an error, you can comment [these lines](https://github.com/Automattic/studio/blob/27f6e6bcb1b7baf8035d5ec794a2a303bb396ffb/src/hooks/use-fetch-wpcom-sites.tsx#L49-L57) connect a simple site, then change click on pull.

> [!NOTE]  
> The pull feature will work in all sites where the import process works.







<img width="1012" alt="Screenshot 2024-11-13 at 14 10 03" src="https://github.com/user-attachments/assets/65b69459-bdf4-4c39-9666-d0c1d186db7b">
<img width="1012" alt="Screenshot 2024-11-13 at 14 09 54" src="https://github.com/user-attachments/assets/9e58cb90-29d5-4602-a351-32bf14a4eb0c">
<img width="1012" alt="Screenshot 2024-11-13 at 15 15 19" src="https://github.com/user-attachments/assets/59ef137a-8e2e-4abf-af43-a48b2a7dbc4f">




## Pre-merge Checklist
